packages/flutter_tools/lib/src/android/application_package.dart

```diff
@@ -116,7 +116,11 @@ class AndroidApk extends ApplicationPackage implements PrebuiltApplicationPackag
     }
 
     if (androidProject.isUsingGradle && androidProject.isSupportedVersion) {
-      apkFile = getApkDirectory(androidProject.parent).childFile(filename);
+      Directory apkDirectory = getApkDirectory(androidProject.parent);
+      if (androidProject.parent.isModule && buildInfo?.mode == BuildMode.debug) {
```

Comment:
What was the PR that changed this? And can you explain why we only do this extra namespace for debug modules?

Reply:
This issue was caused by https://github.com/flutter/flutter/pull/113622, which stopped copying the generated apk to a fixed file name. Module/add2app projects were always outputting the apk in a buildmode subdirectory, eg `<buildmode>/app-<buildmode>.apk`. Previously, the copy to app.apk masked this `<buildmode>/app-<buildmode>.apk` output by placing the apk in `app.apk` without the subdir. When the copy was removed, the search path is now missing a `<buildmode>` subdir only for module projects.

Full flutter apps don't experience this issue due to the output dir not including the buildmode subdir.

We should also do this for the other buildmodes.